core/src/main/java/io/ably/lib/util/JsonUtils.java

```diff
@@ -30,7 +30,10 @@ public JsonUtilsObject add(String key, Object value) {
                 json.addProperty(key, (Number) value);
             } else if (value instanceof JsonUtilsObject) {
                 json.add(key, ((JsonUtilsObject) value).toJson());
+            } else {
+                throw new IllegalArgumentException("Unsupported Object Type");
```

Comment:
Please add a javadoc to this method that would say it throws this exception :wink: Something like
```java
/**
 * Adds the key-value pair to the JSON object. 
 * If the value type is unsupported an IllegalArgumentException is thrown.
 */
```

Reply:
done https://github.com/ably/ably-java/pull/729/commits/196314e6e2539b470618beaa0760426349112141